src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -259,13 +259,19 @@ public function beforeRedirect(Event $event, $url, Response $response)
         if (is_array($url)) {
             $url = Router::url($url + ['_base' => false]);
         }
+        $query = [];
+        if (strpos($url, '?') !== false) {
+            list($url, $querystr) = explode('?', $url);
```

Comment:
Sure but not limiting means that data is lost. The second `?` is interpreted by `parse_str()` fine.

```
php > $a = [];
php > parse_str('a=b&b=c?d=e', $a);
php > var_dump($a);
array(2) {
  ["a"]=>
  string(1) "b"
  ["b"]=>
  string(5) "c?d=e"
}
```
I agree that its a bit weird to have `?` in a key/value, but I've seen some odd stuff over the years.